modules/libraries/src/backend/utils.ts

```diff
@@ -0,0 +1,178 @@
+  const moduleDir = process.LOADED_MODULES['libraries']
+  const nodeFolder = process.pkg ? 'node_production_modules' : 'node_modules'
+
+  if (isOffline) {
+    args.push('--offline')
+  }
+
+  // Hides superfluous messages
+  args.push('--no-fund')
+
+  // Necessary for post install scripts when running from the binary
+  args.push('--scripts-prepend-node-path')
+
+  const cleanArgs = args.map(x => sanitizeArg(x))
+
+  const cwd = customLibsDir ? customLibsDir : sharedLibsDir
+  debug('executing npm', { execPath: process.execPath, moduleDir, cwd, args, cleanArgs })
+
+  try {
+    const spawned = spawn(process.execPath, [`${moduleDir}/${nodeFolder}/npm/bin/npm-cli.js`, ...cleanArgs], {
+      cwd,
+      env: {
+        ...process.env,
+        PATH: `${process.env.PATH}:${path.dirname(process.execPath)}`,
+        PKG_EXECPATH: 'PKG_INVOKE_NODEJS'
+      }
+    })
+
+    const resultBuffer: string[] = []
+
+    spawned.stdout.on('data', msg => resultBuffer.push(msg.toString()))
+    spawned.stderr.on('data', msg => resultBuffer.push(msg.toString()))
+
+    await Promise.fromCallback(cb => spawned.stdout.on('close', cb))
+
+    return resultBuffer.join('')
+  } catch (err) {
+    console.error('error ', err)
+  }
+}
+
+export const createNodeSymlink = async () => {
+  const nodePath = path.join(path.dirname(process.execPath), 'node')
+
+  // The symlink is only necessary when running the binary and node is not installed
+  if (process.execPath.endsWith('bp.exe') || process.execPath.endsWith('bp')) {
+    if (!(await fse.pathExists(nodePath))) {
+      await fse.symlink(process.execPath, nodePath)
+    }
+  }
+}
+
+export const syncAllFiles = async (bp: typeof sdk) => {
+  const files = await bp.ghost.forGlobal().directoryListing(LIB_FOLDER, '*.*')
+  await Promise.mapSeries(files, file => copyFileLocally(file, bp))
+}
+
+export const copyFileLocally = async (fileName: string, bp: typeof sdk): Promise<boolean> => {
+  if (!(await bp.ghost.forGlobal().fileExists(LIB_FOLDER, fileName))) {
+    return false
+  }
+
+  try {
+    const fileContent = await bp.ghost.forGlobal().readFileAsBuffer(LIB_FOLDER, fileName)
+    await fse.writeFile(path.join(sharedLibsDir, fileName), fileContent)
+    return true
+  } catch (err) {
+    bp.logger.error(`Couldn't copy locally. ${err}`)
+    return false
+  }
+}
+
+const deleteLibraryArchive = async (filename: string, bp: typeof sdk) => {
+  try {
+    if (await bp.ghost.forGlobal().fileExists(LIB_FOLDER, filename)) {
+      await bp.ghost.forGlobal().deleteFile(LIB_FOLDER, filename)
+    }
+
+    if (await fse.pathExists(path.join(sharedLibsDir, filename))) {
+      await fse.remove(path.join(sharedLibsDir, filename))
+    }
+  } catch (err) {
+    bp.logger.warn(`Error while deleting the library archive ${err}`)
+  }
+}
+
+export const removeLibrary = async (name: string, bp: typeof sdk): Promise<boolean> => {
+  const packageContent = JSON.parse(await fse.readFile(packageJsonPath, 'UTF-8'))
```

Comment:
any chances for any reason package json to be corrupted or invalid json ?
can you wrap this in try catch so it doesn't crash out 

also why not use [`fse.readJson`](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/readJson.md) ?

Reply:
Ah, didn't knew that existed :)